exporter/elasticsearchexporter/model.go

```diff
@@ -247,6 +254,9 @@ func encodeLogAttributesECSMode(document *objmodel.Document, attrs pcommon.Map,
 			}
 
 			document.AddAttribute(ecsKey, v)
+			if _, ok := preserveMap[k]; ok {
+				document.AddAttribute(k, v)
```

Comment:
nit: isn't it a bit too optimistic to rely on the keys from this map? In this case the keys (or 1 key for now) come from SemConv names and we actually map to ECS. In this specific case it works out fine, since both are named `host.name`, but that's not guaranteed. This could potentially store things in SemConv format on the document which may not match ECS in the future.

ps: Don't see this as a blocker.

Reply:
Hmm, I don't get what is the concern here 🤔 .

At the moment all SemConv attributes that have an ECS equivalent (where ECS_name!=SemConv_name) are converted based on the Map we define at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/961e16b3f29b1661687ac6d862455125306bcdfb/exporter/elasticsearchexporter/model.go#L37.

So the goal of this PR is to provide the option to mark some SemConv attributes from this conversion map which we want to be preserved along with being copied to the ECS equivalent.

So the cases are:
1) rename and keep only ECS
2) copy and keep both

So to your comment, the `preserveMap` makes sense to be checked only if a key is part of the `conversionMap`. Let me know if I miss sth here.

Reply:
> So the goal of this PR is to provide the option to mark some SemConv attributes from this conversion map which we want to be preserved along with being copied to the ECS equivalent.

Yeah, I think that does the job, so that's why I don't see this as a blocker.

What I raise is what if the SemConv and ECS names don't match? (which is not the case now). Because ultimately, we still map to ECS, where we have 2 target fields in ECS and one happens to have the same name as SemConv. So I'd think that in this case the matching `host.name` is a fortunate case.

Don't want to get philosophical, but I'd say we actually don't preserve the SemConv name, but still map it, in this case to the same ECS name, so `host.name` is mapped to `host.name`. This makes zero difference now, but may be relevant where we map from SemConv `x.y` to ECS `foo.x` and `foo.y`. That'd be a 3. case in the list.

But you are right, what needs to be handled for this case, is handled properly. So happy to close this thread.

Reply:
Thank's for elaborating here! I see the point. Maybe a mapping schema like `SemConv_attr -> [ECS_field1, ECS_field2]` would be more appropriate for that case. But I think we can revisit in the future if we see the need.